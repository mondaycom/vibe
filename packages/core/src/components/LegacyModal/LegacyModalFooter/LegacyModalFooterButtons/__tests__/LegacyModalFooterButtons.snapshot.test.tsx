import { describe, it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import ModalFooterButtons from "../LegacyModalFooterButtons";

describe("ModalFooterButtons", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(<ModalFooterButtons primaryButtonText="Confirm" secondaryButtonText="Cancel" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly with no tertiary", () => {
    const tree = renderer.create(<ModalFooterButtons primaryButtonText="Confirm" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly with disabled primary", () => {
    const tree = renderer
      .create(
        <ModalFooterButtons primaryButtonText="Confirm" secondaryButtonText="Cancel" disablePrimaryButton={true} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
