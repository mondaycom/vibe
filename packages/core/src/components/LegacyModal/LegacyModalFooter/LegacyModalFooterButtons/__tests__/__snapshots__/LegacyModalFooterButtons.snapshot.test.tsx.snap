// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalFooterButtons should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="container directionRow justifyEnd alignCenter"
    style={
      {
        "gap": "var(--spacing-small)",
      }
    }
  >
    <button
      aria-busy={false}
      aria-disabled={false}
      className="button sizeMedium kindTertiary colorPrimary"
      data-testid="button"
      data-vibe="Button"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      type="button"
    >
      Cancel
    </button>
    <button
      aria-busy={false}
      aria-disabled={false}
      className="button sizeMedium kindPrimary colorPrimary"
      data-testid="button"
      data-vibe="Button"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      type="button"
    >
      Confirm
    </button>
  </div>
</div>
`;

exports[`ModalFooterButtons should render correctly with disabled primary 1`] = `
<div
  className="container"
>
  <div
    className="container directionRow justifyEnd alignCenter"
    style={
      {
        "gap": "var(--spacing-small)",
      }
    }
  >
    <button
      aria-busy={false}
      aria-disabled={false}
      className="button sizeMedium kindTertiary colorPrimary"
      data-testid="button"
      data-vibe="Button"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      type="button"
    >
      Cancel
    </button>
    <button
      aria-busy={false}
      aria-disabled={true}
      className="button sizeMedium kindPrimary colorPrimary disabled"
      data-testid="button"
      data-vibe="Button"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      tabIndex={-1}
      type="button"
    >
      Confirm
    </button>
  </div>
</div>
`;

exports[`ModalFooterButtons should render correctly with no tertiary 1`] = `
<div
  className="container"
>
  <div
    className="container directionRow justifyEnd alignCenter"
    style={
      {
        "gap": "var(--spacing-small)",
      }
    }
  >
    <button
      aria-busy={false}
      aria-disabled={false}
      className="button sizeMedium kindPrimary colorPrimary"
      data-testid="button"
      data-vibe="Button"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      type="button"
    >
      Confirm
    </button>
  </div>
</div>
`;
